File: zaoch2018-2019/H/ok.cpp
#include <set>
#include <cstdio>
#include <ctime>
#include <vector>
#include <map>
#include <iostream>
#include <string>
#include <stack>
#include <queue>
#include <math.h>
#include <random>
#include <algorithm>
#include <functional>
#include <cassert>
#include <cstring>
#include <list>
#include <bitset>
#include <deque>
#include <complex>
#include <cmath>

using namespace std;
#define x first
#define y second
#define ll long long
#define ld long double

const int MN = 1e5;

int res(int a)
{
    int res = 1;
    while (a)
    {
        res *= a % 10;
        a /= 10;
    }
    return res;
}

int32_t main()
{
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int l, r;
    cin >> l >> r;
    int opt = l;
    for (int i = l; i < r + 1; ++i)
    {
        if (res(opt) < res(i))
        {
            opt = i;
        }
    }
    cout << opt << '\n';
}

File: zaoch2018-2019/H/ok.py
ll = input().strip()
rr = input().strip()

n = len(rr)

l = [0 for i in range(n)]
r = [0 for i in range(n)]

for i in range(n):
    r[i] = int(rr[i])

for i in range(len(ll)):
    l[n - i - 1] = int(ll[len(ll) - i - 1])

def tn(a):
    s = ''.join(map(str, a))
    return int(s)
def gv(a):
    res = 1
    for i in a:
        res *= i
    return res

def clc(a, pos):
    c = a[::]
    for i in range(len(a)):
        # if i == pos:
            # r[i] -= 1
        if i > pos:
            c[i] = 9
    while c[0] == 0:
        c = c[1:]
    return c

ln = int(ll)
rn = int(rr)

opt = r[::]
for i in range(n):
    if r[i] > 0:
        r[i] -= 1
        k = clc(r, i)
        if ln <= tn(k):
            if gv(k) > gv(opt):
                opt = k
        r[i] += 1

print(''.join(map(str, opt)))
